pyomo/contrib/mindtpy/algorithm_base_class.py

```diff
@@ -2740,6 +2699,107 @@ def initialize_mip_problem(self):
+
+        self.set_appsi_solver_update_config()
+
+        if config.mip_solver == 'gurobi_persistent' and config.single_tree:
+            # PreCrush: Controls presolve reductions that affect user cuts
+            # You should consider setting this parameter to 1 if you are using callbacks to add your own cuts.
+            self.mip_opt.options['PreCrush'] = 1
+            self.mip_opt.options['LazyConstraints'] = 1
+
+        # set thread
+        if config.threads > 0:
+            self.mip_opt.options['threads'] = config.threads
+        if config.regularization_mip_threads > 0:
+            self.regularization_mip_opt['threads'] = config.threads
+
+        # regularization solver
+        if config.mip_regularization_solver == 'cplex':
+            if config.solution_limit is not None:
+                self.regularization_mip_opt.options[
+                    'mip limits solutions'
+                ] = config.solution_limit
+            self.regularization_mip_opt.options['mip strategy presolvenode'] = 3
+            # TODO: need to discuss if this option should be added.
+            if config.add_regularization in {'hess_lag', 'hess_only_lag'}:
+                self.regularization_mip_opt.options['optimalitytarget'] = 3
+        elif config.mip_regularization_solver == 'gurobi':
+            if config.solution_limit is not None:
+                self.regularization_mip_opt.options[
+                    'SolutionLimit'
+                ] = config.solution_limit
+            self.regularization_mip_opt.options['Presolve'] = 2
+
+    def set_appsi_solver_update_config(self):
+        """Set update config for APPSI solvers."""
+        config = self.config
+        if config.mip_solver in {'appsi_cplex', 'appsi_gurobi', 'appsi_highs'}:
+            # mip main problem
+            self.mip_opt.update_config.check_for_new_or_removed_constraints = True
+            self.mip_opt.update_config.check_for_new_or_removed_vars = True
+            self.mip_opt.update_config.check_for_new_or_removed_params = False
+            self.mip_opt.update_config.check_for_new_objective = True
+            self.mip_opt.update_config.update_constraints = True
+            self.mip_opt.update_config.update_vars = True
+            self.mip_opt.update_config.update_params = False
+            self.mip_opt.update_config.update_named_expressions = False
+            self.mip_opt.update_config.update_objective = False
+            self.mip_opt.update_config.treat_fixed_vars_as_params = True
+
+        if config.nlp_solver == 'appsi_ipopt':
+            # fixed-nlp
+            self.nlp_opt.update_config.check_for_new_or_removed_constraints = False
+            self.nlp_opt.update_config.check_for_new_or_removed_vars = False
+            self.nlp_opt.update_config.check_for_new_or_removed_params = False
+            self.nlp_opt.update_config.check_for_new_objective = False
+            # TODO: we will deactivate trivial constraints, do we need to update constraints?
```

Comment:
Why would we need to?

Reply:
I think we need to update the constraints. Otherwise, the trivial constraint will remain active.